tests/test_gatk_spark.yml

```diff
@@ -6,24 +6,45 @@
     - preprocessing
   files:
     - path: results/csv/markduplicates.csv
+      md5sum: 0d6120bb99e92f6810343270711ca53e
     - path: results/csv/markduplicates_no_table.csv
+      md5sum: 2a2d3d4842befd4def39156463859ee3
     - path: results/csv/recalibrated.csv
+      md5sum: 42628ec994c16f565e5407b40a9c1ac3
     - path: results/multiqc
     - path: results/preprocessing/markduplicates/test/test.md.cram
+      md5sum: 7a1ff0ec3f9150dd6f10174037c79a8a
     - path: results/preprocessing/markduplicates/test/test.md.cram.crai
+      md5sum: fd96e4bb9c861965c090a54877057908
     - path: results/preprocessing/recal_table/test/test.recal.table
+      md5sum: 5b6e5078b4a90f6cb982fa0f0df616c2
     - path: results/preprocessing/recalibrated/test/test.recal.cram
+      md5sum: e61d1e30c33276dd4458f0372da38f9a
     - path: results/preprocessing/recalibrated/test/test.recal.cram.crai
+      md5sum: e1647d01bff62853ac5d8f79795b10ac
     - path: results/reports/fastqc/test-test_L1
     - path: results/reports/markduplicates/test/test.md.metrics
+    # The text-based output-file test.md.metrics only contains comments and no metrics, which is a bit odd.
```

Comment:
This is the content of `results/reports/markduplicates/test/test.md.metrics` when I run the test `nextflow run main.nf -profile test,use_gatk_spark,singularity --save_output_as_bam` on my server:

```
## htsjdk.samtools.metrics.StringHeader
# EstimateLibraryComplexity --INPUT test.bam --OUTPUT test.md.metrics --TMP_DIR . --REFERENCE_SEQUENCE genome.fasta --MIN_IDENTICAL_BASES 5 --MAX_DIFF_RATE 0.03 --MIN_MEAN_QUALITY 20 --MAX_GROUP_RATIO 500 --MAX_READ_LENGTH 0 --MIN_GROUP_COUNT 2 --READ_NAME_REGEX <optimized capture of last three ':' separated fields as numeric values> --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 2 --MAX_RECORDS_IN_RAM 6839119 --CREATE_INDEX false --CREATE_MD5_FILE false --GA4GH_CLIENT_SECRETS client_secrets.json --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false
## htsjdk.samtools.metrics.StringHeader
# Started on: Wed Aug 10 18:59:28 GMT 2022



```
I'm using singularity while on GitHub the CI-tests are being executed using Docker. It might be a combination of my server, gatk-spark and singularity that causes the file to be empty. Unfortunately, the file can't be tested with an md5-check as the file contains a timestamp.

Reply:
Can it be tested with a `contains` of the header?

Reply:
I could test for the presence of the header/comment, but is it relevant? I don't know.  If the file contain some metrics, I would, of course, try to test for those.

Reply:
I forgot about this, we should check it out, to make sure it works fine

Reply:
So I again ran the test
```
nextflow run main.nf -profile test,use_gatk_spark,singularity --save_output_as_bam
```
 And again the resulting metrics-file `results/reports/markduplicates/test/test.md.metrics` just contained the comment mentioned above. I also ran `picard EstimateLibraryComplexity` manually in the appropriate work-folder, and I got the same metrics-file with any metrics 🤔
 
 This was the command I ran:
 ```
 picard EstimateLibraryComplexity --INPUT test.bam --OUTPUT test2.md.metrics --TMP_DIR . --REFERENCE_SEQUENCE genome.fasta --MIN_IDENTICAL_BASES 5 --MAX_DIFF_RATE 0.03 --MIN_MEAN_QUALITY 20 --MAX_GROUP_RATIO 500 --MAX_READ_LENGTH 0 --MIN_GROUP_COUNT 2  --OPTICAL_DUPLICATE_PIXEL_DISTANCE 100 --MAX_OPTICAL_DUPLICATE_SET_SIZE 300000 --VERBOSITY INFO --QUIET false --VALIDATION_STRINGENCY STRICT --COMPRESSION_LEVEL 2 --MAX_RECORDS_IN_RAM 6839119 --CREATE_INDEX false --CREATE_MD5_FILE false --GA4GH_CLIENT_SECRETS client_secrets.json --help false --version false --showHidden false --USE_JDK_DEFLATER false --USE_JDK_INFLATER false
 ```